From 55475b89552fa68147f0895a1501854960e6bb6c Mon Sep 17 00:00:00 2001
From: technomancer <dominikbehr@yahoo.com>
Date: Thu, 30 Jan 2014 23:33:22 -0800
Subject: [PATCH] failure to set drm master is not fatal

When running as user, it is not possible to call drmSetMaster.
But the first DRM client gets the master automatically, so it
is not necessary to set it.

Signed-off-by: Dominik Behr <dbehr@chromium.org>
---
 src/radeon_kms.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/radeon_kms.c b/src/radeon_kms.c
index 21a420f..3133c8d 100644
--- a/src/radeon_kms.c
+++ b/src/radeon_kms.c
@@ -1171,8 +1171,8 @@ Bool RADEONScreenInit_KMS(SCREEN_INIT_ARGS_DECL)
 
     ret = drmSetMaster(info->dri2.drm_fd);
     if (ret) {
-        ErrorF("Unable to retrieve master\n");
-        return FALSE;
+	xf86DrvMsg(pScrn->scrnIndex, X_WARNING,
+		   "Unable to retrieve master\n");
     }
     info->directRenderingEnabled = FALSE;
     if (info->r600_shadow_fb == FALSE)
@@ -1394,7 +1394,7 @@ Bool RADEONEnterVT_KMS(VT_FUNC_ARGS_DECL)
 
     ret = drmSetMaster(info->dri2.drm_fd);
     if (ret)
-	ErrorF("Unable to retrieve master\n");
+	xf86DrvMsg(pScrn->scrnIndex, X_WARNING, "Unable to retrieve master\n");
     info->accel_state->XInited3D = FALSE;
     info->accel_state->engineMode = EXA_ENGINEMODE_UNKNOWN;
 
-- 
1.8.3.2

